lib/Doctrine/MongoDB/Aggregation/Expr.php

```diff
@@ -0,0 +1,890 @@
+     * @param mixed|self $expression
+     * @return self
+     */
+    public function minute($expression)
+    {
+        return $this->operator('$minute', $expression);
+    }
+
+    /**
+     * Divides one number by another and returns the remainder. The first argument is divided by the second argument.
+     *
+     * The arguments can be any valid expression as long as they resolve to numbers.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/mod/
+     * @param mixed|self $expression1
+     * @param mixed|self $expression2
+     * @return self
+     */
+    public function mod($expression1, $expression2)
+    {
+        return $this->operator('$mod', array($expression1, $expression2));
+    }
+
+    /**
+     * Returns the month of a date as a number between 1 and 12.
+     *
+     * The argument can be any expression as long as it resolves to a date.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/month/
+     * @param mixed|self $expression
+     * @return self
+     */
+    public function month($expression)
+    {
+        return $this->operator('$month', $expression);
+    }
+
+    /**
+     * Multiplies numbers together and returns the result.
+     *
+     * The arguments can be any valid expression as long as they resolve to numbers.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/multiply/
+     * @param mixed|self $expression1
+     * @param mixed|self $expression2
+     * @param mixed|self $expression3,... Additional expressions
+     * @return self
+     */
+    public function multiply($expression1, $expression2 /* , $expression3, ... */)
+    {
+        return $this->operator('$multiply', func_get_args());
+    }
+
+    /**
+     * Compares two values and returns:
+     * true when the values are not equivalent.
+     * false when the values are equivalent.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/ne/
+     * @param mixed|self $expression1
+     * @param mixed|self $expression2
+     * @return self
+     */
+    public function ne($expression1, $expression2)
+    {
+        return $this->operator('$ne', array($expression1, $expression2));
+    }
+
+    /**
+     * Evaluates a boolean and returns the opposite boolean value.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/not/
+     * @param mixed|self $expression
+     * @return self
+     */
+    public function not($expression)
+    {
+        return $this->operator('$not', $expression);
+    }
+
+    /**
+     * Defines an operator and value on the expression.
+     *
+     * If there is a current field, the operator will be set on it; otherwise,
+     * the operator is set at the top level of the query.
+     *
+     * @param string $operator
+     * @param array|self[]|self $expression
+     * @return self
+     */
+    protected function operator($operator, $expression)
```

Comment:
`operator()` is public in the basic query builder. Would making this public remove the need for the `expression()` method you have above? That would allow ultimate flexibility if the aggregate framework added more top-level operators (like `$and`), which do not require a field.